File: frontend/src/modules/mod_stats/mod_stats.js
import api from '@/core/api.js'

export function setupStats(store, actions) {
    return {
        // 调度模块一：描述性宏观统计
        async runDescriptiveStats() {
            if (store.showStats) { store.showStats = false; return; }
            if (store.selectedVars.length === 0) return actions.showDialog({ title: '提示', message: '请勾选需要进行基础统计的数值变量！' });
            actions.addLog("正在向分析中台投递描述性统计计算指令...");
            try {
                const res = await api.post('/api/analyze/descriptive', {
                    filename: store.currentDataFile, columns: store.selectedVars
                });
                if (res.data.status === 'success') {
                    store.statsResult = res.data.data;
                    store.showStats = true;
                    actions.addLog("基础描述统计测度计算完成！", "success");
                }
            } catch (err) { actions.addLog(`❌ 统计失败: ${err.message}`, "error"); }
        }
    };
}

File: backend/modules/mod_stats/mod_stats.py
import os
import pandas as pd
from flask import Blueprint, request, jsonify
from core.config import UPLOAD_FOLDER
from core.utils import read_df

bp = Blueprint('stats', __name__)


@bp.route('/api/analyze/descriptive', methods=['POST'])
def descriptive_analysis():
    """获取标量特征维度的描述性综合测度"""
    try:
        df = read_df(os.path.join(UPLOAD_FOLDER, request.json.get('filename')))
        selected = [c for c in request.json.get('columns', []) if c in df.columns]
        stats_data = []
        for col in selected:
            d = df[col].dropna()
            stats_data.append({
                "variable": str(col), "count": int(d.count()), "mean": float(round(d.mean(), 4)),
                "median": float(round(d.median(), 4)), "std": float(round(d.std(), 4)),
                "min": float(round(d.min(), 4)), "max": float(round(d.max(), 4))
            })
        return jsonify({"status": "success", "data": stats_data})
    except Exception as e:
        return jsonify({"status": "error", "message": str(e)}), 500
